src/System.Net.HttpListener/tests/GetContextHelper.cs

```diff
@@ -60,7 +60,15 @@ public class Helpers
     {
         public static void WaitForSocketShutdown(Socket socket)
         {
-            while (SocketConnected(socket));
+            if (PlatformDetection.IsWindows)
```

Comment:
These shenanigans are needed because the native implementation is dumb and doesn't know when a socket is closed, but the managed implementation uses sockets directly so this is fine